tests/integration/study_data_blueprint/test_binding_constraints.py

```diff
@@ -53,6 +56,191 @@ def test_lifecycle__nominal(self, client: TestClient, user_access_token: str) ->
+        clusters_list = res.json()
+        assert res.status_code == 200, res.json()
+        assert len(clusters_list) == 1
+        assert clusters_list[0]["name"] == "cluster1"
+        assert clusters_list[0]["group"] == "Nuclear"
+
+        # Create two binding constraints in the study
+        res = client.post(
+            f"/v1/studies/{study_id}/bindingconstraints",
+            json={
+                "name": "binding_constraint_1",
+                "enabled": True,
+                "time_step": "hourly",
+                "operator": "less",
+                "coeffs": {},
+                "comments": "",
+            },
+            headers=user_headers,
+        )
+        assert res.status_code == 200, res.json()
+
+        res = client.post(
+            f"/v1/studies/{study_id}/bindingconstraints",
+            json={
+                "name": "binding_constraint_2",
+                "enabled": True,
+                "time_step": "hourly",
+                "operator": "less",
+                "coeffs": {},
+                "comments": "",
+            },
+            headers=user_headers,
+        )
+        assert res.status_code == 200, res.json()
+
+        # Get binding constraints list to check created binding constraints
+        res = client.get(f"/v1/studies/{study_id}/bindingconstraints", headers=user_headers)
+        binding_constraints_list = res.json()
+        assert res.status_code == 200, res.json()
+        assert len(binding_constraints_list) == 2
+        assert binding_constraints_list[0]["id"] == "binding_constraint_1"
+        assert binding_constraints_list[1]["id"] == "binding_constraint_2"
+
+        bc_id = binding_constraints_list[0]["id"]
+
+        # Add binding constraint link term
+        res = client.post(
+            f"/v1/studies/{study_id}/bindingconstraints/{bc_id}/term",
+            json={
+                "weight": 1,
+                "offset": 2,
+                "data": {"area1": area1_name, "area2": area2_name},
+            },
+            headers=user_headers,
+        )
+        assert res.status_code == 200, res.json()
+
+        # Add binding constraint cluster term
+        res = client.post(
+            f"/v1/studies/{study_id}/bindingconstraints/{bc_id}/term",
+            json={
+                "weight": 1,
+                "offset": 2,
+                "data": {"area": area1_name, "cluster": "cluster1"},
+            },
+            headers=user_headers,
+        )
+        assert res.status_code == 200, res.json()
+
+        # Get binding constraints list to check added terms
+        res = client.get(
+            f"/v1/studies/{study_id}/bindingconstraints/{bc_id}",
+            headers=user_headers,
+        )
+        binding_constraint = res.json()
+        constraints = binding_constraint["constraints"]
+        assert res.status_code == 200, res.json()
+        assert binding_constraint["id"] == bc_id
+        assert len(constraints) == 2
```

Comment:
We should rather check the list of terms of the constraint by comparing directly with a list of dictionaries.
```python
binding_constraint = res.json()
constraints = binding_constraint["constraints"]
expected = [
    {"data": {"area1": area1_id, "area2": area2_id}, "id": "area1%area2", "offset": 2.0, "weight": 1.0},
    {"data": {"area": area1_id, "cluster": cluster_id}, "id": transform_name_to_id(f"area1.{cluster_id}"), "offset": 2.0, "weight": 1.0},
]
assert constraints == expected
```

Again, there is a confusion between names and IDs (which are converted in lower case for the binding constraints).